airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -290,7 +290,7 @@ def run(
         autocommit: bool = False,
         parameters: Optional[Union[Sequence[Any], Dict[Any, Any]]] = None,
         handler: Optional[Callable] = None,
-    ):
+    ) -> Optional[List]:
```

Comment:
Is there any use case for empty `sql` argument? Maybe it would be better to raise `ValueError` instead of doing unnecessary connection opening. WDYT?